libmamba/tests/src/specs/test_match_spec.cpp

```diff
@@ -128,6 +128,19 @@ namespace
             REQUIRE(ms.str() == "abc>3");
         }
 
+        SECTION("numpy~=1.26.0")
+        {
+            auto ms = MatchSpec::parse("numpy~=1.26.0").value();
+            REQUIRE(ms.name().str() == "numpy");
+            REQUIRE(ms.version().str() == "~=1.26.0");
+            REQUIRE(ms.build_string().is_explicitly_free());
+            REQUIRE(ms.build_number().is_explicitly_free());
+            REQUIRE(ms.str() == "numpy~=1.26.0");
+
+            auto ms2 = MatchSpec::parse(ms.str()).value();
+            REQUIRE(ms2 == ms);
```

Comment:
This is a first test the assumption described in https://github.com/mamba-org/mamba/pull/3651/files#r1952499642.